### DISCUSSION TITLE
My client cannot connect to the auth server in any way

### DISCUSSION THREAD

**Discussion by NicoloLazzaroni** [DarkflameUniverse/DarkflameServer#1290] [Q&A] [answered]
### Make sure you've done the following:

- [X] I have checked that this issue has not already been reported.

- [X] I have validated that this issue is not a syntax error of either MySQL or SQLite.

- [X] I have pulled the latest version of the main branch of DarkflameServer and have confirmed that the issue exists there.

### DarkflameServer Version

latest

### Issue Description

Hello,

I have been trying to setup DLU on my dedicated server but I always run into the same problem.

When the server is started, my client cannot pass the login screen with the "we could not log you in..." message.
Nothing appears in the console.

This is very weird because of all the steps tried here:
- I have tried with different clients, different OS for the player, with Wireshark sniffing and from the client the UDP packages are leaving correctly.
- I tried different networks.
- The firewall is okay, been tested multiple times with two methods:
   a. The Docker version works, and I can login (packets reaching the server) (even if it is unplayable due to many bugs like getting stuck at 35 and 55 percent).
   b. I used https://check-host.net/check-udp?lang=en to check if the port 1001 was open and in the console I received multiple RakNet empty package errors (packets reaching the server).
- The external IP seems correct, as the Docker version works.
- tcpdump shows the UDP packages reaching the dedicated server correctly, strangely not DLU.

I also:
- Reinstalled client multiple times.
- Reinstalled DLU multiple times.
- Reset DB.
- The client is configured correctly (also proven by the fact that I can connect to the Docker version).

The server starts correctly, no errors on the console.

My network configuration is, however, a bit specific. I use an OVH VPS for outgoing and incoming connection to gain from their antiddos.
`PUBLIC IP [OVH] <----GRE----> 10.0.0.1 [OVH] <----GRE----> 10.0.0.2 [Router] <----GRE----> Router public IP <----Router DHCP----> 192.168.255.1 <----Router DHCP----> 192.168.255.11 [Server]`
Even tho I don't see how it could be the problem as the docker version works.

The server is NOT on my own network.

I also tried 192.168.255.11 as EXTERNAL_IP, still nothing.

Master IP is set to `127.0.0.1` (the only one that works).

Any ideas?

Thank you for your time.

### Reproduction Steps

Just a normal setup following all the instructions, nothing in particular.

### Expected Behavior

Should allow me to login.

### Environment

Linux - Debian 12 - Dedicated Server.

**Comment by EmosewaMC** [comment]
are you forwarding port 1001?  That or an incorrect IP are one of the few reasons you cannot connect to auth.  Moving to discussion as this is not a widespread dlu issue.

**Comment by NicoloLazzaroni** [comment]
Yes, I am forwarding port 1001 and, as said in the main text, packets can reach the server on that port, and empty packets reach DLU causing RakNet’s empty packet console error.

**Reply by EmosewaMC** [reply]
The empty packets are not reaching dlu at all, evidenced by the empty packet console errors; I am not sure what you are doing to setup your server, but this has not been an issue for over 2 years now, and has never popped up except in this specific case.  The only thing that may be related is #225, but other than that there isnt anything we can do but debug RakNet.

**Reply by NicoloLazzaroni** [reply]
Only the packets from https://check-host.net/check-udp?lang=en show the empty packet error, because they are empty with no RakNet data in them.
Normal packets from LU client just show no error.

I am happy to debug RakNet, but cannot find anything in your docs about how to do it.

I cannot understand what issue #225 is about as I can see the server binds to 0.0.0.0, not to the external IP.

**Reply by NicoloLazzaroni** [reply]
Moreover, I had the same issue with the first release of DLU 1+ year ago, I just thought it was a launch bug that would get quickly solved and did not report it.

**Reply by EmosewaMC** [reply, marked as answer]
Well of course we don't have docs on how to debug raknet, it's a thirdparty library we don't own and we are using it specifically so client compatibility is there.  We don't really have to debug raknet that often especially for an issue no one on the dev team can reproduce.  I'm able to setup my aws hosted server, other server owners have their gcp or hetzner setups and they all work, except this one.  Unless you have some way to setup this very specific network configuration simply there isn't anything we can do to help.  Dlu has does not officially support so much tunneling and we don't test for it as it's just not worth the time and effort to set all this up.

**Reply by NicoloLazzaroni** [reply]
No problem. I could make the Dockerized version work properly with the TCP-shim, so even if the main version one doesn't I can survive. It would have been interesting to understand why tho, but we'll never know.
